### PR TITLE
Default DB + parsing

### PR DESCRIPTION
**Pull request by AsafMah**
### Added
- Added Initial Catalog (Default Database) parameter to ConnectionStringBuilder
- Added method to manually set the cache for CloudSettings
### Changed
- Urls with one item after the path (i.e https://test.com/abc) will now be treated as cluster and initial catalog (ie. the cluster is "https://test.com" and the initial catalog is "abc").
    * This is to align our behaviour with the .NET SDK
### Fixed
- Some edge cases in url parsing

### CODE REVIEW

File: azure-kusto-data/azure/kusto/data/client_base.py [hunk 1]
```diff
@@ -58,6 +61,11 @@ def __init__(self, kcsb: Union[KustoConnectionStringBuilder, str], is_async):
         self.client_details = self._kcsb.client_details
         self._is_closed: bool = False
 
+        self.default_database = self._kcsb.initial_catalog
+
+    def get_database(self, database_name: str) -> str:
```
Comment (yogilad) [inline azure-kusto-data/azure/kusto/data/client_base.py:66]: This can be a property as in KCSB

**Review by yogilad — APPROVED** [review]
Approved with some suggestions

File: azure-kusto-data/azure/kusto/data/client_base.py (hunk 1 above)
Comment (AsafMah) [reply azure-kusto-data/azure/kusto/data/client_base.py:66]: There is a "inital_catalog" property in kcsb, that's only a small helper for client
It should probably be private too
